groth16-framework/tests/common/erc20.rs

```diff
@@ -0,0 +1,99 @@
+//! Test utilities for ERC20 query
+
+use super::{TestContext, TestQuery, L};
+use ethers::prelude::U256;
+use groth16_framework::{test_utils::save_plonky2_proof_pis, C, D, F};
+use mr_plonky2_circuits::{
+    api::{deserialize_proof, serialize_proof, ProofWithVK},
+    block::PublicInputs as BlockDbPublicInputs,
+    query_erc20::{
+        block::BlockPublicInputs, revelation::RevelationRecursiveInput, RevelationErcInput,
+    },
+    utils::{Packer, ToFields},
+};
+use mrp2_utils::utils::convert_u256_to_u32_fields;
+use plonky2::{
+    field::types::{Field, PrimeField64, Sample},
+    hash::hash_types::HashOut,
+    plonk::proof::ProofWithPublicInputs,
+};
+
+impl<const BLOCK_DB_DEPTH: usize> TestContext<BLOCK_DB_DEPTH> {
+    /// Generate a fake ERC20 query proof.
+    pub(crate) fn generate_erc20_query_proof(
+        &self,
+        output_dir: &str,
+        query: &TestQuery,
+        block_db_proof: &ProofWithPublicInputs<F, C, D>,
+        query_result: U256,
+    ) -> Vec<u8> {
+        // Generate a fake ERC20 query proof.
+        let block_db_pi = BlockDbPublicInputs::<F>::from(&block_db_proof.public_inputs);
+        let query_max_number = block_db_pi.block_number_data() - F::ONE;
```

Comment:
it's just in the test for the ERC20 in the revelation (that Steven probably copied here since it's exactly the same syntax), i use the latest block db - 1 as the upper bound of the query .

Reply:
Fixed in commit [ea7ca54](https://github.com/Lagrange-Labs/Euclid-database/pull/21/commits/ea7ca548c1b0ceea776c23a74ebba98b5ed438c9).